src/interop/js-interop/js-types.md

```diff
@@ -0,0 +1,242 @@
+---
+title: JS types
+description: Usage information about the core types in JS interop.
+---
+
+Dart values and JS values belong to separate language domains. When compiling to
+Wasm, they execute in separate *runtimes* as well. As such, you should treat JS
+values as foreign types. To provide Dart types for JS values,
+[`dart:js_interop`] exposes a set of [extension types] prefixed with `JS` called
```

Comment:
nit: I think the fact that they are extension-types are an implementation detail that we can delay to explain later. Connecting this with the comment below, I think if we leave out the "representation type" detail from the next paragraph, but explain it more with the lens of them being inconsistent and unreified, then we can probably also drop here the fact that they are currently implemented as extension-types.

Reply:
Agreed that we can remove the notion of extension types and representation types and just refer to inconsistent runtime types.